### DISCUSSION TITLE
How to redirect traffic from subdomain to root

### DISCUSSION THREAD

**Discussion by konradsienkowski** [nuxt/nuxt#16534] [Questions]
Hey everyone,

I'm working on a new page for a project which currently has two parts - one on the subdomain (docs.domain.com), and the other on the root domain. We'd like to have both of them combined in one Nuxt app. 

Since there are many links pointing to the subdomain, I want to redirect all the traffic coming to the subdomain (`docs.*`) to specific routes within the root domain (e.g. `/docs/*`).

Tried to find a solution using [global middleware](https://v3.nuxtjs.org/guide/directory-structure/middleware), but those seem to work only on the client-side and both of their arguments (to, from) do not include a hostname to detect the subdomain traffic.

If anyone has any ideas or examples on that, I'd be really glad. Thanks!

**Comment by sevillaarvin** [comment]
I don't have a solution for Nuxt but if I were you I would handle this with a proxy in front of Nuxt that will handle the redirect, like nginx.

**Comment by TheAlexLichter** [comment]
See https://github.com/nuxt/framework/discussions/6057

**Reply by Apadolsky** [reply]
This link is referenced in a few discussions about subdomains, can you please provide an updated link to this reference as it results in a 404. - Thanks

**Reply by TheAlexLichter** [reply]
@Apadolsky already in contact with the GitHub support so they can redirect the discussion properly 👍 

In the meantime, have a look at https://github.com/zernonia/keypress/blob/main/app/router.options.ts as example.
